static/js/narrow_state.js

```diff
@@ -352,3 +352,27 @@ export function is_for_stream_id(stream_id) {
 
     return stream_id === narrow_sub.stream_id;
 }
+
+export const allowed_web_public_narrows = [
+    "streams",
+    "stream",
+    "topic",
+    "sender",
+    "has",
+    "search",
+    "near",
+    "id",
+];
+
+export function is_web_public_compatible(ops) {
+    // When modifying this function, update similar function in narrow.py too.
+    for (const op of ops) {
+        if (op.operand === undefined) {
+            return false;
+        }
+        if (!allowed_web_public_narrows.includes(op.operator)) {
+            return false;
+        }
+    }
+    return true;
+}
```

Comment:
I think this function belongs in the `Filter` class, which also has an existing test suite that's better designed for this sort of question.  (Correct me if I'm wrong).

Reply:
Done.